pex/environment.py

```diff
@@ -651,6 +655,20 @@ def _declare_namespace_packages(cls, resolved_dists):
                 )
                 for index, (dist, ns_packages) in enumerate(namespace_packages_by_dist.items())
             )
+            if not pkg_resources:
```

Comment:
This change and many test fixes revolve around the full switch to `importlib.resources` in Python 3.12. The setuptools maintainers are killing `pkg_resources` as of Python 3.12 and `pkg_resources`, like Pex, relies on import hooks to enable vendored code it uses. The setuptools did not update those hooks to work in a Python 3.12 world (Pex had to in #2170); so even when `pkg_resources` is present, it fails to import under 3.12.